src/AddressRegistry.Consumer.Read.Municipality/Infrastructure/Program.cs

```diff
@@ -41,98 +39,162 @@ public static async Task Main(string[] args)
-                .SetBasePath(Directory.GetCurrentDirectory())
-                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: false)
-                .AddJsonFile($"appsettings.{Environment.MachineName.ToLowerInvariant()}.json", optional: true, reloadOnChange: false)
-                .AddEnvironmentVariables()
-                .AddCommandLine(args)
-                .Build();
+            Log.Information("Starting AddressRegistry.Consumer.Read.Municipality");
 
-            var container = ConfigureServices(configuration);
+            var host = new HostBuilder()
+                .ConfigureAppConfiguration((_, builder) =>
+                {
+                    builder
+                        .SetBasePath(Directory.GetCurrentDirectory())
+                        .AddJsonFile("appsettings.json", optional: false, reloadOnChange: false)
+                        .AddJsonFile($"appsettings.{Environment.MachineName.ToLowerInvariant()}.json", optional: true, reloadOnChange: false)
+                        .AddEnvironmentVariables()
+                        .AddCommandLine(args);
+                })
+                .ConfigureLogging((hostContext, builder) =>
+                {
+                    SelfLog.Enable(Console.WriteLine);
+
+                    Log.Logger = new LoggerConfiguration()
+                        .ReadFrom.Configuration(hostContext.Configuration)
+                        .Enrich.FromLogContext()
+                        .Enrich.WithMachineName()
+                        .Enrich.WithThreadId()
+                        .Enrich.WithEnvironmentUserName()
+                        .Destructure.JsonNetTypes()
+                        .CreateLogger();
+
+                    builder.ClearProviders();
+                    builder.AddSerilog(Log.Logger);
+                })
+                .ConfigureServices((hostContext, services) =>
+                {
+                    var loggerFactory = new SerilogLoggerFactory(Log.Logger);
+
+                    services
+                        .AddTransient(s => new TraceDbConnection<MunicipalityConsumerContext>(
+                            new SqlConnection(hostContext.Configuration.GetConnectionString("ConsumerMunicipality")),
+                            hostContext.Configuration["DataDog:ServiceName"]))
+                        .AddDbContextFactory<MunicipalityConsumerContext>((provider, options) => options
+                            .UseLoggerFactory(loggerFactory)
+                            .UseSqlServer(provider.GetRequiredService<TraceDbConnection<MunicipalityConsumerContext>>(),
+                                sqlServerOptions =>
+                                {
+                                    sqlServerOptions.EnableRetryOnFailure();
+                                    sqlServerOptions.MigrationsHistoryTable(MigrationTables.ConsumerReadMunicipality, Schema.ConsumerReadMunicipality);
+                                }), ServiceLifetime.Transient);
```

Comment:
@ArneD we have added a `ServiceLifetime lifetime` parameter and set to `Transient`. By default is was `Singleton`. We think it should also be added in the streetname consumer.

Reply:
I don't think this should be done, if I read this correct the `ContextFactory` is now transient, while it really should be a singleton. The context it resolves should be transient.